### PR TITLE
Improvements to ocp-indent-compat and the Janestreet profile

### PR DESCRIPTION
**Pull request by Julow**

### CODE REVIEW

**Comment by gpetiot** [comment]
Did we get some feedback from Jane Street on this branch?

**Comment by Julow** [comment]
Yes. This contains a few patches that should be upstreamed as well as https://github.com/ocaml-ppx/ocamlformat/pull/2371, which requires some work.

A first batch was merged in https://github.com/ocaml-ppx/ocamlformat/pull/2428 and just a few patches remain here.

**Comment by Julow** [comment]
Changes to the default profile remain but they all seem to be fixes. I'll submit them as separate PRs to consider them in isolation.

File: test/passing/tests/js_source.ml [hunk 1]
```diff
@@ -8034,6 +8023,19 @@ type t =
...
+    {|
+         f|}
+*)
+
+type t =
+  { xxxxxxxxxxxxxxxxxxx : yyy
+                          [@zzzzzzzzzzzzzzzzzzz
+                             (* ________________________________
+                                ___ *)
+                            _______]
```
Comment (gpetiot) [inline test/passing/tests/js_source.ml:8036]: Shouldn't the multi-line attribute make the first line break?

**Comment by Julow** [comment]
No diff with test_branch on the conventional profile, let's merge !

**Pull request merged**